src/main/java/seedu/address/logic/parser/delivery/DeliveryEditCommandParser.java

```diff
@@ -47,8 +47,19 @@ public DeliveryEditCommand parse(String args) throws ParseException {
         argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_CUSTOMER_ID,
                 PREFIX_DATE, PREFIX_STATUS, PREFIX_NOTE);
 
-        DeliveryEditDescriptor deliveryEditDescriptor =
-                new DeliveryEditDescriptor();
+        DeliveryEditDescriptor deliveryEditDescriptor = createDeliveryEditDescriptor(argMultimap);
+
+        return new DeliveryEditCommand(index, deliveryEditDescriptor);
+    }
+
+    /**
+     * Creates and returns a {@code DeliveryEditDescriptor} based on the given {@code ArgumentMultimap}.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public DeliveryEditDescriptor createDeliveryEditDescriptor(ArgumentMultimap argMultimap) throws ParseException {
+
+        DeliveryEditDescriptor deliveryEditDescriptor = new DeliveryEditDescriptor();
 
         if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
```

Comment:
You may want to extract out like `argMultimap.getValue(PREFIX_NAME);` to `Optional<Name> name = argMultimap.getValue(PREFIX_NAME);` so you can just:

`if (name.isPresent()) { ... } `
and you can easily just do `name.get()`

Looks cleaner.